app/routes/cache.tsx

```
@@ -19,7 +20,29 @@ export const loader = async ({request}: LoaderFunctionArgs) => {
     return json([] as string[], {status: 401})
   }
 
-  return await loadCacheKeys()
+  const keys = await loadCacheKeys()
+
+  // Pre-fetch titles for article-like keys (short keys with only alphanumeric chars)
+  const articleKeys = keys.filter((key) => /^[A-Z0-9]{4}$/.test(key))
+
+  // Load cached values and extract titles
+  const titles: Record<string, string> = {}
+
+  for (const key of articleKeys) {
+    try {
+      const value = await loadCacheValue(key)
+      if (value) {
+        const parsed = JSON.parse(value)
+        if (parsed && parsed.data && parsed.data.title) {
+          titles[key] = parsed.data.title
+        }
+      }
+    } catch (e) {
+      // Ignore errors for individual keys
```

not sure if await inside for loop is a good idea, perhaps it's fine since this code runs on CF and waits from KV store, would need to give it a bit more time to think through..

but I don't agreee with using try/catch with empty catch 🤔

---

The try/catch is because there could be false-positives in matching articleKeys, I wanted to ignore those. What would be a cleaner way to do this?

Since it's dev-only, I don't really care about speed, but if you think it's a problem I can just remove this title-fetching feature.

---

> there could be false-positives in matching articleKeys

if that is a bug, please report an issue (or what's the number of existing issue?) 🙏 
if not a bug and/or if it is about typos in GDocs and we already have some script to report the problems in Discord server or elsewhere, please add a note in code comments how the exceptions are handled elsewhere so we don't need to do anything about the false positives here

---

I merged but we can still discuss this.

What I meant is that some keys (such as _allQuestions_ and _redirects_) do not correspond to article keys, and I am only interested in the latter. Filtering for `/^[A-Z0-9]{4}$/` mostly fixes that, but in theory I could imagine something like *FEAT* that is not an article, but that fits the regex, and the code is to exclude these. It might make more sense to check if these keys have a *data->title* attribute and ignore them if they don't instead of a try/catch though.

Since I merged, I notice that https://aisafety.info/cache takes 5+ seconds to load (it's instant on dev), and this is probably the cause. I think it's a good enough reason to remove this feature (titles with keys) entirely.

---

ah, I see, thanks for the explanation - code comments are ideal place to explain reasons *why* some code is there, e.g. `// Ignore errors for individual keys` is completely useless comment but `// Ignore errors for keys that don't match any articles, like allQuestions and redirects, but not filtered out by the above regex` would be perfect for this kind of documentation IMO

yeah, seems really slow in production and best to undo after all 😢